mlir/tools/rocmlir-tuning-driver/rocmlir-tuning-driver.cpp

```diff
@@ -330,30 +331,11 @@ benchmarkKernels(ArrayRef<std::string> binaries,
     }
   });
 
-  // Warmup run
+  // Warmup run. We measure the warmup to get an estimate of the kernel runtime.
+  // We will use this estimate to determine if the kernel is small or not.
+  float totalMillisecondsWarmup = 0.0;
+  HIPCHECK(hipStreamSynchronize(stream));
```

Comment:
Why do we have to synchronize here?

Reply:
The events we use for measuring are passed to `hipExtModuleLaunchKernel` which should be implemented in a way that only captures the runtime of the given kernel, presumably by recording the events on the same stream. Something like:
```
hipEventRecord(startEvent, stream);
launchKernel(stream);
hipEventRecord(stopEvent, stream);
```
So, any preceding hip calls shouldn't be captured by the events.